### ISSUE TITLE
How to write Markdown links that reference local webserver root path

### ISSUE THREAD

**Issue by Industry-Standard** [arnaudsm/raito#52]
@arnaudsm, Is it possible to do the equivalent of ``<a href="/">Home</a>`` or ``<a href="/subdir/a.html">Goto A</a>`` in Markdown in Raito?

I've tried many variations but had no luck, and all the examples (/subdir/a, subdir/b, etc) use relative backpathing, e.g. ``[home](../../..)`` in Markdown to get back to the homepage.  

Any server root links do not work in markdown, e.g.

``` markdown
[Next](/dir1/) 

[Next](/dir1/README.md)

[Home](/)
```

Thanks for your time!

**Comment by arnaudsm** [comment]
It should be feasible without too much refactoring. That logic is located [here](https://github.com/arnaudsm/raito/blob/f549621d5159ad52bdfef7060df1f17c5e426a23/index.html#L351)

Sorry I have little spare time these days, contributions are welcome ! I might work on this later

**Comment by mckaygerhard** [comment]
hi @Industry-Standard  this can be workaround by using ceo friendly urls.. 

this is not the best cos means your index.html code and all the document will be hard coded with the configure site.. 

with the current behaviour you can move the files and will work with any other markdown engine.. 

by example my wiki site works in the raito logic and also in the github and forgetto logic wiki.. so means that current behaviour is full compatible with any engine or renderer.. 

your request is a fashion tendency of assume that root are always the root and is not the case.. making the logic incompatible and unportable.. 

imagine i must move 200 files made in raito to a test environemnt.. in your case i must setup a domain and a root dir only to test.. with current case is just put the files and define a new variable of root in the index.html

**Comment by mckaygerhard** [comment]
hi @Industry-Standard  this can be workaround by using ceo friendly urls.. 

this is not the best cos means your index.html code and all the document will be hard coded with the configure site.. 

with the current behavior you can move the files and will work with any other markdown engine.. 

by example my wiki site works in the raito logic and also in the github and forgetto logic wiki.. so means that current behavior is full compatible with any engine or render.. 

your request is a fashion tendency of assume that root are always the root and is not the case.. making the logic incompatible and unportable.. 

imagine i must move 200 files made in raito to a test environment.. in your case i must setup a domain and a root dir only to test.. with current case is just put the files and define a new variable of root in the index.html

another reason of current behavior is that you can avoid rewrite crap making fully portable to any web server

**Comment by Industry-Standard** [comment]
tl;dr - I have this working here: https://github.com/Industry-Standard/raitoo/tree/relative-path-handling

You can now use the following with ``config.browserRouter`` on or off:

* absolute links (e.g. "http://site.com/thing")
* root-relative links (e.g. "/docs/subdir/a")
* directory-relative links (e.g. "../docs/", "./subdir/a", "docs/subdir/b.md")

Root relative links will use ``config.sitePath`` and javascript ``window.location`` to determine what raito's *root* path is. 

These changes introduce a few extra features that I felt were necessary:

* ``config.serveLocalNonMdContent``: allows raito to directly link to media (images, videos, etc) and have it correctly redirect to those files on the server. E.G. click on a markdown link to an image, get the image (this wasn't working before; you could *reference* images for marked to render, but not link to images)
* ``config.serveDirectories``: choose whether or not raito will link to a raw directory (that doesn't have a README.md) on your server - this is still a WIP on my branch, but almost done
* ``config.indexFilename``: defaults to "README", but can be something else
* ``config.indexExtension``: defaults to ".md", but could be something like ".mdx" or whatever

All that being said, I don't think this changes the _philosophy_ of what raito does, but it is a large change to the actual _codebase_. I've merged these into master on my fork without issue, but I fully understand if there is not an interest to introduce these changes into upstream Raito. They are there if someone wants to try and merge them upstream. 

Thanks for the great work on raito - it's been fun to hack on it!
